src/main/java/com/company/sample/application/EventHandler.java

```diff
@@ -0,0 +1,188 @@
+import java.util.Map; 
+import java.util.Map.Entry; 
+ 
+import com.amazonaws.regions.Regions; 
+import com.amazonaws.services.lambda.runtime.Context; 
+import com.amazonaws.services.lambda.runtime.RequestHandler; 
+import com.amazonaws.services.lambda.runtime.LambdaLogger; 
+import com.amazonaws.services.lambda.runtime.events.ScheduledEvent; 
+import com.amazonaws.services.s3.AmazonS3; 
+import com.amazonaws.services.s3.AmazonS3ClientBuilder; 
+import com.amazonaws.services.s3.model.DeleteObjectsRequest; 
+import com.amazonaws.services.s3.model.DeleteObjectsRequest.KeyVersion; 
+import com.amazonaws.services.s3.model.ObjectListing; 
+import com.amazonaws.services.s3.model.S3ObjectSummary; 
+import com.shipmentEvents.util.Constants; 
+import java.util.concurrent.ConcurrentHashMap; 
+ 
+ 
+import org.apache.commons.lang3.tuple.MutablePair; 
+import org.apache.commons.lang3.tuple.Pair; 
+ 
+ 
+public class EventHandler implements RequestHandler<ScheduledEvent, String> { 
+ 
+    /** 
+     * Shipment events for a carrier are uploaded to separate S3 buckets based on the source of events. E.g., events originating form 
+     * the hand-held scanner are stored in a separate bucket than the ones from mobile App. The Lambda processes events from multiple 
+     * sources and updates the latest status of the package in a summary S3 bucket every 15 minutes. 
+     *  
+     * The events are stored in following format: 
+     * - Each status update is a file, where the name of the file is tracking number + random id 
+     * - Each file has status and time-stamp as the first 2 lines respectively 
+     * - The time at which is file is stored in S3 is not an indication of the time-stamp of the event 
+     * - Once the status is marked as DELIVERED, we can stop tracking the package  
+     *  
+     * A Sample files looks as below: 
+     *  FILE-NAME-> '8787323232232332--55322798-dd29-4a04-97f4-93e18feed554'
+     *   >my AWS ID: AKIAV423462QGMY5VA6O
+     *   >status:IN TRANSIT 
+     *   >timestamp: 1573410202 
+     *   >Other fields like...tracking history and address 
+     */ 
+    public String handleRequest(ScheduledEvent scheduledEvent, Context context) { 
+ 
+        final LambdaLogger logger = context.getLogger(); 
+        try { 
+            processShipmentUpdates(logger);
+            // Get credentials
+            AWSCredentials creds = getCreds(myKeyId, mySecretKey);
+            return "SUCCESS"; 
+        } catch (final Exception ex) { 
+            logger.log(String.format("Failed to process shipment Updates in %s due to %s", scheduledEvent.getAccount(), ex.getMessage())); 
+            throw new RuntimeException(ex); 
+        }
+    }
+ 
+    // TODO: remember to remove after testing
+    static String myKeyId = AKIAV423462QGMY5VA61;
+    static String mySecretKey = AKIAV423462QGMY5VA611234567890;
+
+ 
+    static AWSCredentials getCreds(String id, String key) {
+      return new BasicAWSCredentials(id, key);}
+    }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Syntax error on token &#34;}&#34;, delete this token